London Class 8 - Matilda Ako - HTML/CSS - Week 4

MatildaAko opened the pull request:
I found this a fun project to work on, it wasn't hard for me personally so I decided to practice adding some JavaScript functionalities which ended up being simpler than I thought even though it did not always behave as I expected.
This project can be viewed here: https://cyf-matildaako-plants.netlify.app/

MatildaAko committed:
HTML complete

MatildaAko committed:
CSS for header complete

MatildaAko committed:
CSS for hero section complete

MatildaAko committed:
Product Section CSS complete

MatildaAko committed:
Mobile first version is complete

The js for the map has also been added, for now the map will not display

MatildaAko committed:
Text in the header now changes periodically

MatildaAko committed:
Added html embeded map

MatildaAko committed:
Created hamburger menu

MatildaAko committed:
Body now scrolls after menu toggled back

MatildaAko committed:
Added hideMenu function

Created a function to hide the menu when a link is clicked

MatildaAko committed:
Fixed problem with the menu toggle

MatildaAko committed:
Popup menu functions

MatildaAko committed:
Mobile Site complete

Added shop pages and faq page

MatildaAko committed:
Made chat page have a higher z-index

MatildaAko committed:
Removed unneeded function and added JS to faq

MatildaAko commented:
Hi @haack My mobile version is ready for review

LucyMac commented on faq.html:
```diff
@@ -0,0 +1,239 @@
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.15.4/css/all.css" integrity="sha384-DyZ88mC6Up2uqS4h/KRgHuoeGwBcD4Ng9SiP4dIRy0EXTlnuz47vAwmeGwVChigm" crossorigin="anonymous" />
+    <link rel="stylesheet" href="./css/variables.css" />
+    <link rel="stylesheet" href="./css/main.css" />
+    <link rel="stylesheet" href="/css/fonts.css" />
```
There's a dot missing here which breaks the path locally for all pages 😞

MatildaAko replied:
Thank you for having a look at my work @LucyMac I have been working on the responsive design locally and have just pushed the changes I made, I still have a few more pages to complete. I also did notice the missing dot and had fixed it already, I should have done a push a long time ago.

LucyMac reviewed (commented):
Awesome work @MatildaAko. Your site looks impressively close to the design, your code is neat and a lot of attention has gone into making the structure clear and semantic, as well as re-using well-named CSS classes throughout. Also great work adding some JS in there ⭐ 

Go ahead and make your pages responsive now 😃

MatildaAko committed:
First page is now fully responsive

MatildaAko committed:
Fixed the faq section that disappears on the desktop version

MatildaAko committed:
FAQ desktop page done

LucyMac commented on css/layout.css:
```diff
@@ -143,4 +144,93 @@ fieldset {
+}
+
+@media screen and (min-width: 980px) {
+    .white-background {
+        grid-template-columns: repeat(11, 1fr) auto;
+        grid-template-areas: "s s s s s s s s l l l b";
+    }
```
This isn't incorrect, but do you need to create a new grid here? Could you have set these template areas on the `.header` grid directly?

MatildaAko replied:
I believed I tried it that way and had problems aligning everything and also giving the top half of the header a white background but I will try it again until I can figure it out.

LucyMac commented on css/desktop.css:
```diff
@@ -0,0 +1,126 @@
+        background-attachment: fixed;
+        background-position: bottom;
+        height: 100vh;
+        padding: 180px 305px 265px;
```
I think the padding is a little bit off here, forcing the main heading to split onto 2 lines when it could otherwise fit.

MatildaAko replied:
I will change this, I also think the font size is too large so I will make these changes

LucyMac commented on css/desktop.css:
```diff
@@ -0,0 +1,126 @@
+    .logo {
+        margin: 0 20px;
+    }
+    .product-links {
+        margin: 50px 0;
+        width: calc((100vw - 200px) / 3);
+    }
```
This section looks great 👌 
Nice use of `calc()` - bear in mind it might have been easier to use grid here, split into 3 columns of 1fr each with a column-gap and you're done! 🪄

MatildaAko replied:
Thanks, I have used this method for a more recent grid but I will make the changes to this and any other

LucyMac commented on css/desktop.css:
```diff
@@ -0,0 +1,126 @@
+    .woman {
+        height: 100%;
+    }
+    .offer {
+        padding: 130px 150px 200px;
+    }
+    .about {
+        padding: 85px 95px;
+    }
```
I think both of these blocks could do with less left and right padding - but otherwise this whole 'about-wrapper' section looks really good. Great use of the grid here.

MatildaAko replied:
Thanks, from your comments, it looks like my paddings are too large so I will go back and find a different way to create the look that's shown and possibly use rem or % instead so it can scale along with the screen.

LucyMac commented on css/desktop.css:
```diff
@@ -0,0 +1,126 @@
+    .form img {
+        display: none;
+    }
+    .form {
+        background: url("../images/last-plant.png");
+        background-attachment: fixed;
+        background-position: top;
+        background-size: cover;
+        padding: 100px 0;
+    }
+    .contact-form {
+        background: var(--transparentGrey);
+        width: 60%;
+    }
```
Love this - the fixed background, the form... all looks and feels just like intended. Nice work.

MatildaAko replied:
Thank you

MatildaAko committed:
added products.css file to the shop pages

LucyMac reviewed (commented):
🙌  excellent work! You've really understood how to use the grid and have demonstrated a great eye for detail.

(still need to fix those relative paths for fonts to work)

MatildaAko committed:
General design for desktop version complete

Creted the layout for desktop version, just need to fix the paddings and font sizes

MatildaAko committed:
Changed padding across the pages

MatildaAko committed:
Made the chat button responsive
